js/src/modals/SMSVerification/store.js

```diff
@@ -0,0 +1,236 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { observable, computed, autorun, action } from 'mobx';
+import phone from 'phoneformat.js';
+import { sha3 } from '../../api/util/sha3';
+
+import ABI from '../../contracts/abi/sms-verification.json';
+// TODO: move this to a better place
+const contract = '0xcE381B876A85A72303f7cA7b3a012f58F4CEEEeB';
+
+import checkIfVerified from './check-if-verified';
+import checkIfRequested from './check-if-requested';
+import checkIfTxFailed from './check-if-tx-failed';
+import waitForConfirmations from './wait-for-confirmations';
+import postToVerificationServer from './post-to-verification-server';
+
+const validCode = /^[A-Z\s]+$/i;
+
+export const GATHERING_DATA = 'gathering-data';
+export const GATHERED_DATA = 'gathered-data';
+export const POSTING_REQUEST = 'posting-request';
+export const POSTED_REQUEST = 'posted-request';
+export const REQUESTING_SMS = 'requesting-sms';
+export const REQUESTED_SMS = 'requested-sms';
+export const QUERY_CODE = 'query-code';
+export const POSTING_CONFIRMATION = 'posting-confirmation';
+export const POSTED_CONFIRMATION = 'posted-confirmation';
+export const DONE = 'done';
+
+export default class VerificationStore {
+  @observable step = null;
+  @observable error = null;
+
+  @observable fee = null;
+  @observable isVerified = null;
+  @observable hasRequested = null;
+  @observable consentGiven = false;
+  @observable number = '';
+  @observable requestTx = null;
+  @observable code = '';
+  @observable confirmationTx = null;
+
+  @computed get isCodeValid () {
+    return validCode.test(this.code);
+  }
+  @computed get isNumberValid () {
+    return phone.isValidNumber(this.number);
+  }
+
+  @computed get isStepValid () {
+    if (this.step === DONE) {
+      return true;
+    }
+    if (this.error) {
+      return false;
+    }
+
+    if (this.step === GATHERED_DATA) {
+      return this.fee && this.isVerified === false && this.isNumberValid && this.consentGiven;
+    }
+    if (this.step === REQUESTED_SMS) {
+      return this.requestTx;
+    }
+    if (this.step === QUERY_CODE) {
+      return this.isCodeValid;
+    }
+    if (this.step === POSTED_CONFIRMATION) {
+      return this.confirmationTx;
+    }
+    return false;
+  }
+
+  @action setNumber = (number) => {
+    this.number = number;
+  }
+  @action setConsentGiven = (consentGiven) => {
+    this.consentGiven = consentGiven;
+  }
+  @action setCode = (code) => {
+    this.code = code;
+  }
+
+  constructor (api, account) {
```

Comment:
Following from what we've done in the React classes, I always put the variables first (`@observable`), then the initialisation (`constructor`) then the actual `@computed` values (functions) then the actual `@action` (interact) & then the internals.

Valid question, suggestions welcome.